### PR TITLE
Upgrading @kitajs/html and @kitajs/ts-html-plugin

### PR DESCRIPTION
**Pull request by n3op2**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [x] Bug Fix
- [x] Feature

## Linked tickets

https://digicatapult.atlassian.net/browse/VR-187

## High level description

Upgrading @kitajs and required changes so typescript compiles .tsx files using this lib

## Detailed description

Updated @kitajs modules to address renovate errors as per story mentioned above. This also includes changes to the other files to address the deprecations

## Operational impact

Medium - might have impact on rendering htmx however, tried locally and it has rendered homepage with no issues

## Additional context

Have had to ts-ignore some un-used variables

### CODE REVIEW

File: src/controllers/queries/__tests__/helpers.ts [hunk 1]
```diff
@@ -43,9 +43,13 @@ function templateListFake(templateName: string, ...args: any[]) {
 export const withQueriesMocks = (stage: Scope3FormStage = 'companySelect', compNumber: string = '10000009') => {
   const scope3CarbonConsumptionTemplateMock = {
     newScope3CarbonConsumptionFormPage: (
+      // @ts-ignore
```
Comment (mattdean-digicatapult) [inline src/controllers/queries/__tests__/helpers.ts:46]: why? Can we not fix whatever the warnings were?

File: src/views/common.tsx [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import { Html } from '@kitajs/html'
+import type { PropsWithChildren } from '@kitajs/html'
+import { escapeHtml } from '@kitajs/html'
```
Comment (mattdean-digicatapult) [inline src/views/common.tsx:2]: why the change in import? It's now generating warnings

Comment (n3op2) [reply src/views/common.tsx:2]: Because it's no longer under `{ Html }` and also mentioned in https://github.com/kitajs/docs/pull/36/files#diff-4e4ca54733d4a35f3b755176b22a0cca067a01ad2e9b28f94c1881544a221c0f deprecations so was just following their suggestions. The `import { Html }` generates warning as per screenshot below:
Screenshot 2024-07-31 at 13 07 46

So imported PropsWithChildren as a type and escapeHtml as it's, open to suggestions.

File: src/controllers/queries/__tests__/helpers.ts (hunk 1 above)
Comment (n3op2) [reply src/controllers/queries/__tests__/helpers.ts:46]: unused variable

File: src/views/common.tsx (hunk 2 above)
Comment (mattdean-digicatapult) [reply src/views/common.tsx:2]: Screenshot 2024-07-31 at 11 13 54

File: src/controllers/queries/__tests__/helpers.ts (hunk 1 above)
Comment (mattdean-digicatapult) [reply src/controllers/queries/__tests__/helpers.ts:46]: if we don't use the variable can we just rmeove it?

File: src/views/common.tsx (hunk 2 above)
Comment (n3op2) [reply src/views/common.tsx:2]: this should be reflected in the commit.

File: src/controllers/queries/__tests__/helpers.ts (hunk 1 above)
Comment (n3op2) [reply src/controllers/queries/__tests__/helpers.ts]: it has been removed and should be reflect in the file changes upon next push
